packages/astro/src/runtime/client/dev-overlay/plugins/utils/highlight.ts

```diff
@@ -0,0 +1,50 @@
+import type { DevOverlayHighlight } from '../../ui-library/highlight.js';
+import type { Icon } from '../../ui-library/icons.js';
+
+export function createHighlight(rect: DOMRect, icon?: Icon) {
+	const highlight = document.createElement('astro-overlay-highlight') as DevOverlayHighlight;
+	if (icon) highlight.icon = icon;
+
+	highlight.tabIndex = 0;
+
+	positionHighlight(highlight, rect);
+	return highlight;
+}
+
+export function positionHighlight(highlight: DevOverlayHighlight, rect: DOMRect) {
+	// Make an highlight that is 10px bigger than the element on all sides
+	highlight.style.top = `${Math.max(rect.top + window.scrollY - 10, 0)}px`;
+	highlight.style.left = `${Math.max(rect.left + window.scrollX - 10, 0)}px`;
+	highlight.style.width = `${rect.width + 15}px`;
+	highlight.style.height = `${rect.height + 15}px`;
+}
+
+export function attachTooltipToHighlight(
+	highlight: DevOverlayHighlight,
+	tooltip: HTMLElement,
+	originalElement: Element
+) {
+	highlight.shadowRoot.appendChild(tooltip);
```

Comment:
just so you know, all modern browsers support `append()` so you can save a few keystrokes